wrangler-core/src/main/java/io/cdap/directives/column/SetHeader.java

```diff
@@ -90,5 +92,14 @@ public Mutation lineage() {
       .generate(Many.of(columns))
       .build();
   }
-}
 
+  @Override
+  public Schema getOutputSchema(SchemaResolutionContext context) {
+    List<Schema.Field> fields = new ArrayList<>(context.getInputSchema().getFields());
+    for (int i = 0; i < columns.size(); i++) {
+      Schema.Field existing = fields.remove(i);
+      fields.add(i, Schema.Field.of(columns.get(i), existing.getSchema()));
```

Comment:
1. Will there be a case where number of elements in `columns` are greater than number of elements in `inputSchema`? in that case, `existing.getSchema()` might throw a `NullPointerException`?
2. Why are we removing and adding elements to existing input schema rather than creating a new list? if we can, then we should for readability.